### PR TITLE
add build tools and rpms from build hosts to koji metadata, use build…

### PR DESCRIPTION
**Pull request by rcerven**
… hostname as buildroot_id

* CLOUDBLD-10926

Signed-off-by: Robert Cerven <rcerven@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [x] Python type annotations added to new code
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [x] Changes to metadata also update the documentation for the metadata
- n/a Pull request has a link to an osbs-docs PR for user documentation updates
- n/a New feature can be disabled from a configuration file

### CODE REVIEW

**Comment by rcerven** [comment]
requires https://github.com/containerbuildsystem/osbs-client/pull/1113 to be merged

File: atomic_reactor/plugins/gather_builds_metadata.py [hunk 1]
```diff
@@ -54,6 +57,56 @@ def _generate_build_log_output(self, platform: str, buildroot_id: int) -> Option
...
+
+        for hostname, attr in platform_config.items():
+            if build_host != hostname:
+                continue
+
+            remote_host = RemoteHost(
+                hostname=hostname, username=attr["username"], ssh_keyfile=attr["auth"],
+                slots=attr.get("slots", 1), socket_path=attr["socket_path"], slots_dir=slots_dir
+            )
+            break
```
Comment (mkosiarc) [inline atomic_reactor/plugins/gather_builds_metadata.py:97]: this whole for loop seems redundant to me. Can't we simply do something like this?
```
if build_host in platform_config: # maybe even this check is redundant
    attr = platform_config[build_host]
    remote_host = RemoteHost(
                hostname=build_host, username=attr["username"], ssh_keyfile=attr["auth"],
                slots=attr.get("slots", 1), socket_path=attr["socket_path"], slots_dir=slots_dir
            )

```
This way we avoid the for loop, ugly continue and break commands.

File: atomic_reactor/plugins/gather_builds_metadata.py [hunk 2]
```diff
@@ -54,6 +57,56 @@ def _generate_build_log_output(self, platform: str, buildroot_id: int) -> Option
...
+
+        for task_name in task_results:
+            if not (task_name.startswith('binary-container-build') and task_platform in task_name):
+                continue
+
+            if 'task_result' not in task_results[task_name]:
+                raise RuntimeError(f"task_results is missing from: {task_name}")
+
+            build_host = json.loads(task_results[task_name]['task_result'])
+            break
```
Comment (mkosiarc) [inline atomic_reactor/plugins/gather_builds_metadata.py:73]: even here we have a for loop, continue and break statements when looking for one element, it does not seem readable or pythonic. I would prefer something like:
```
task_name = next(filter(lambda task_name: task_name.startswith('binary-container-build'), task_results))
...rest of the code
```
It seems more readable, less lines and it has straightforward flow, so you don't have to watch the continue, break statements.

File: atomic_reactor/plugins/gather_builds_metadata.py (hunk 1 above)
Comment (rcerven) [reply atomic_reactor/plugins/gather_builds_metadata.py:97]: ok

File: atomic_reactor/plugins/gather_builds_metadata.py (hunk 2 above)
Comment (rcerven) [reply atomic_reactor/plugins/gather_builds_metadata.py]: ok

**Review by ben-alkov — APPROVED** [review]
LGTM

**Review by mkosiarc — APPROVED** [review]

**Comment by rcerven** [comment]
rn updated

**Pull request merged**